BuildScript/BuildTasks/MetadataValueDictionary.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership.  rubicon licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.  You may obtain a copy of the
+// License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Remotion.BuildScript.BuildTasks
+{
+  /// <summary>
+  /// Wraps a <see cref="IDictionary{TKey,TValue}"/> and performs case-insensitive access operations, emulating default msbuild behavior.
+  /// </summary>
+  public class MetadataValueDictionary
+  {
+    private readonly IReadOnlyDictionary<string, string> _rawDictionary;
+    private readonly StringComparer _dictionaryStringComparer = StringComparer.OrdinalIgnoreCase;
+
+    public MetadataValueDictionary (IDictionary<string, string> rawDictionary)
```

Comment:
Where do we build the dictionary parameter and is that place providing information in case of duplicate keys?

Reply:
It was only implicitly done by the runtime. I've added an explicit check in `CreateTestConfigurations.GetMetadataValueDictionary`